### ISSUE TITLE
[ Question]: Is this line a bug?

### ISSUE THREAD

**Issue by kozer** [meta-llama/llama#1024]
https://github.com/facebookresearch/llama/blob/ef351e9cd9496c579bf9f2bb036ef11bdc5ca3d2/llama/generation.py#L192
I'm currently looking at the code, and trying to get myself around it.
I have a question about the above line.
Shouldn't the `logits` go through a softmax in this line? We want to take the probabilities, is that correct?
Thank you very much!

**Comment by shenxiangzhuang** [comment]
Hi @kozer, I think the answer maybe is: we can do it, but we don't have to. 

Why we can do it? Because in the full & original train/inference processes, we need apply `softmax` before the output of Transformer block and output the *probability* as given in the Transformer paper.

But we don't have to. Especially in the single inference process. We can see the output of the model is *logit*, not *probability*. Because we use the model output only for comparison (to get the max value), so we don't need apply `softmax` due to it's monotonically increasing property.

**Comment by kozer** [comment]
Thank you so much for your responses! Make sense!!
